evm_arithmetization/src/cpu/kernel/interpreter.rs

```diff
@@ -532,8 +428,9 @@ impl<'a, F: Field> Interpreter<'a, F> {
             .set(field.unscale(), value);
     }
 
-    pub(crate) fn get_txn_data(&self) -> &[U256] {
-        &self.generation_state.memory.contexts[0].segments[Segment::TxnData.unscale()].content
+    pub(crate) fn get_txn_data(&self) -> Vec<U256> {
+        self.generation_state.memory.contexts[0].segments[Segment::TxnData.unscale()]
+            .return_content()
```

Comment:
Just `content()` is perhaps a better name? Or `fetch_content()` if you really want to emphasize that it's extracted.